pypesto/history/base.py

```diff
@@ -246,51 +251,117 @@ def get_trimmed_indices(self) -> np.ndarray:
         fval_trace = self.get_fval_trace()
         return np.where(fval_trace <= np.fmin.accumulate(fval_trace))[0]
 
+    def implements_trace(self) -> bool:
+        """Check whether the history has a trace that can be queried."""
+        try:
+            self.get_fval_trace()
+        except NotImplementedError:
+            return False
+
+        return True
 
-class History(HistoryBase):
+
+class NoHistory(HistoryBase):
+    """Dummy history that does not do anything.
+
+    Can be used whenever a history object is needed, but no history is desired.
+    Can be created, but not queried.
     """
-    Tracks number of function evaluations only, no trace.
 
-    Parameters
-    ----------
-    options:
-        History options.
+    def update(  # noqa: D102
+        self,
+        x: np.ndarray,
+        sensi_orders: Tuple[int, ...],
+        mode: ModeType,
+        result: ResultDict,
```

Comment:
```suggestion
        *args, 
        **kwargs,
```
Also for all other not-planned-to-ever-be-implemented methods?

Reply:
I think it makes sense to define the function header in as much detail as possible, right?

Reply:
Fine to keep, but doesn't feel helpful here. args/kwargs is less verbose and won't require changes if the base signature changes.